api/include/mfxstructures.h

```diff
@@ -2269,6 +2269,48 @@ MFX_PACK_END()
 
 #endif
 
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+/* Multi-adapters Querying structs */
+typedef enum
+{
+    MFX_ENCODE_COMPONENT = 1,
+    MFX_DECODE_COMPONENT = 2,
+    MFX_VPP_COMPONENT    = 3
+} mfxComponentType;
+
+MFX_PACK_BEGIN_STRUCT_W_PTR()
+typedef struct
+{
+    mfxComponentType Type;
```

Comment:
hm... We have few reserved fields in mfxVideoParam. Maybe we will just add `mfxComponentType` in there and avoid defining mfxComponentInfo altogether?

Reply:
No, I don't like this idea. We widely use ```mfxVideoParam``` struct in other parts of API and design of this struct assumes that it is component agnostic

Reply:
That's another badly designed API. mfxVideoParam has union inside _without_ any way to understand which union path is to be followed from the data stored in the structure. IMHO, mfxComponentType should have been initially in this union from the msdk birth day!

Reply:
Well, maybe, but MSDK uses different approach for component type resolution. We use different functions calls, i.e. resolution happens on level of function name